packages/user/src/components/UserMenu.tsx

```diff
@@ -14,10 +14,25 @@ const UserMenu = (properties: Properties) => {
   } = properties;
 
   const { user } = useUser();
+  const [emailVerificationEnabled, userEmailVerified] = useEmailVerification();
 
-  return (
-    <>{user?.isEmailVerified ? authenticatedUserMenu : anonymousUserMenu}</>
-  );
+  const renderUserMenu = () => {
+    if (user) {
+      if (!emailVerificationEnabled) {
+        return authenticatedUserMenu;
+      } else {
+        if (!userEmailVerified) {
+          return null;
+        }
+
+        return authenticatedUserMenu;
+      }
+    }
+
+    return anonymousUserMenu;
+  };
```

Comment:
```suggestion
 const renderUserMenu = () => {
    if (!user) {
      return anonymousUserMenu;
    }
  
    if (emailVerificationEnabled && !userEmailVerified) {
      return null;
    }
  
    return authenticatedUserMenu;
};
```
Sometimes we can invert the ifs to make the code simpler.